### PR TITLE
Combine metering-olm and helm-charts and update to new nested OLM manifest bundle layout

### PR DESCRIPTION
**Pull request by chancez**
- First commit is to update our directory layout for the new marketplace publishing pipeline for Red Hat operators.
- Second commit prepares for some other refactorings related to charts coming soon, primarily it combines helm-operator and metering-olm charts, and simplifies our manifest generation process.
- Third commit removes superset chart which is rarely used and only intended for developers, which could be deployed manually anyways.
- Fourth commit passes the images to use for each component as an env var to metering-helm-operator. The marketplace publishing pipeline will substitute these values so our operator can use different images for OCP instead of the hardcoded defaults.

### CODE REVIEW

File: manifests/deploy/openshift/metering-operator/metering-operator-deployment.yaml [hunk 1]
```diff
@@ -54,12 +54,12 @@ spec:
...
-            memory: 100Mi
-          requests:
-            cpu: 50m
-            memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+            requests:
+              cpu: 50m
+              memory: 100Mi
```
Comment (EmilyM1) [inline manifests/deploy/openshift/metering-operator/metering-operator-deployment.yaml]: Is everything indented 1 too far in from 57-62?

Comment (chancez) [reply manifests/deploy/openshift/metering-operator/metering-operator-deployment.yaml]: Yeah, that looks too far.

File: manifests/deploy/openshift/olm/bundle/4.1/reportdatasource.crd.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    plural: reportdatasources
+    singular: reportdatasource
+    kind: ReportDataSource
+    shortNames:
+    - datasource
+    - datasources
+  additionalPrinterColumns:
+  - name: Earliest Metric
+    type: string
+    JSONPath: .status.prometheusMetricImportStatus.earliestImportedMetricTime
```
Comment (timflannagan1) [inline manifests/deploy/openshift/olm/bundle/4.1/reportdatasource.crd.yaml]: I believe this needs to be updated now that #706 was just merged.

File: manifests/deploy/openshift/olm/bundle/4.1/meteringoperator.v4.1.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -85,7 +85,7 @@ metadata:
             "name": "unready-deployment-replicas"
           },
           "spec": {
-            "prometheusMetricsImporter": {
+            "promsum": {
```
Comment (timflannagan1) [inline manifests/deploy/openshift/olm/bundle/4.1/meteringoperator.v4.1.0.clusterserviceversion.yaml]: Should this be here?

File: manifests/deploy/openshift/olm/bundle/4.1/reportdatasource.crd.yaml (hunk 2 above)
Comment (chancez) [reply manifests/deploy/openshift/olm/bundle/4.1/reportdatasource.crd.yaml]: Ah, I should have done that in the other PR. Fixed here.

File: manifests/deploy/openshift/olm/bundle/4.1/meteringoperator.v4.1.0.clusterserviceversion.yaml (hunk 3 above)
Comment (chancez) [reply manifests/deploy/openshift/olm/bundle/4.1/meteringoperator.v4.1.0.clusterserviceversion.yaml]: Ah, yeah, whoops.

**Pull request merged**